### PR TITLE
clang-tidy cleanups

### PR DESCRIPTION
**Pull request by neheb**
small for now.

### CODE REVIEW

File: modules/ldapbackend/ldapbackend.hh [hunk 1]
```diff
@@ -123,16 +123,11 @@ class LdapBackend : public DNSBackend
   {
     QType qtype;
     DNSName qname;
-    uint32_t ttl;
-    time_t lastmod;
+    uint32_t ttl{0};
+    time_t lastmod{0};
```
[the remark this answers is not in the document]
Comment (neheb) [reply modules/ldapbackend/ldapbackend.hh:127]: What type change?

[the remark this answers is not in the document]
Comment (omoerbeek) [reply modules/ldapbackend/ldapbackend.hh:127]: Oops read the diff wrong, sorry

File: pdns/dns.hh [hunk 2]
```diff
@@ -72,18 +72,18 @@ public:
...
   QType qtype; //!< qtype of this record, ie A, CNAME, MX etc
-  uint16_t qclass; //!< class of this record
+  uint16_t qclass{1}; //!< class of this record
 
-  uint8_t scopeMask;
-  bool auth;
-  bool disabled;
+  uint8_t scopeMask{0};
+  bool auth{1};
+  bool disabled{0};
```
Comment (rgacogne) [inline pdns/dns.hh:86]:
```suggestion
  bool auth{true};
  bool disabled{false};
```

**Review by rgacogne — APPROVED** [review]
One nit that would not prevent this PR from being merged, especially since it was already like that before.
clang-tidy is reporting several things that were not introduced by this PR, but we need to decide whether we want to fix them now or open an issue to fix them later.

**Comment by neheb** [comment]
I looked into the clang-tidy errors. Those are a lot of errors to fix...

File: pdns/dns.hh (hunk 2 above)
Comment (neheb) [reply pdns/dns.hh]: Fixed.

**Comment by rgacogne** [comment]
> I looked into the clang-tidy errors. Those are a lot of errors to fix...

Sorry, I should have been more clear about the fact that we do not expect _you_ to fix them :) If we decide to fix them now, do you mind if we push a few commits to this branch?

**Comment by neheb** [comment]
It’s best to handle later then. I need to remove a few commits. Something broke.

**Comment by neheb** [comment]
Fixed.

**Comment by neheb** [comment]
ping @rgacogne

File: pdns/saxfr.cc [hunk 3]
```diff
@@ -129,7 +129,7 @@ try
         throw PDNSException("tcp read failed");
 
       len=ntohs(len);
-      std::unique_ptr<char[]> creply(new char[len]);
+      auto creply = std::make_unique<char[]>(len);
```
Comment (rgacogne) [inline pdns/saxfr.cc:132]: Not your doing but switching to a `std::vector` or a `PacketBuffer` would be nice, perhaps in a follow-up PR.

**Review by rgacogne — APPROVED** [review]

**Review by rgacogne — APPROVED** [review]

**Comment by neheb** [comment]
anything else needed to merge this?

**Comment by rgacogne** [comment]
I was hoping for a second review, but I'll just do a second pass myself.

**Pull request merged**